### DISCUSSION TITLE
Write to localStorage - set-local-storage-item scriptlet

### DISCUSSION THREAD

**Discussion by anewuser** [uBlockOrigin/uBlock-issues#2697] [Ideas] [answered]
Thank you for implementing the `set-cookie` scriptlet  (https://github.com/uBlockOrigin/uBlock-issues/issues/1518#issuecomment-1594263723). 

Are you also going to  add  [`set-local-storage-item`](https://github.com/AdguardTeam/Scriptlets/blob/master/src/scriptlets/set-local-storage-item.js) and  `trusted-set-local-storage-item`? AdGuard has several examples of use cases: [AnnoyancesFilter/Cookies/sections/cookies_specific.txt](https://github.com/AdguardTeam/AdguardFilters/blob/master/AnnoyancesFilter/Cookies/sections/cookies_specific.txt). People can also use these scriptlets to enforce specific options on sites (like dark themes) without having to keep their other localStorage data indefinitely.

**Comment by peace2000** [comment]
That would help to deal with GDPR dialogs. Example on this page: `https://www.is.fi/politiikka/art-2000008974427.html`

These filters would deal with the GDPR dialog so that embedded content in articles won't be broken:

`is.fi##+js(trusted-set-cookie, consentUUID, cb373c5f-140d-489c-81f6-996fb913aa04_20, 1year)`

`is.fi##+js(trusted-set-local-storage-item, cookieConsent, {"sppd":0\,"spcx":0\,"spam":0\,"spma":0\,"spad":0\,"spem":1})`

`is.fi` uses highly generic GDPR script provided by `cdn.privacy-mgmt.com` so it would definetely help on numerous cases.

**Comment by gorhill** [comment, marked as answer]
https://github.com/gorhill/uBlock/commit/9443ba80a06c8748fc5164bb0ad6da752a8b3bf3

**Comment by bruna16** [comment]
https://github.com/uBlockOrigin/uBlock-issues/discussions/2697#discussioncomment-6231761

**Comment by peace2000** [comment]
@gorhill Should we also have a scriptlet to remove local storage items as a safety measure? If ever a local storage item causes issues, it can't be removed just by removing the problematic rule. It would need to be removed from pc's via a scriptlet.

(It could have adblocking uses as well though I don't have a case on my mind.)

**Reply by gorhill** [reply]
<https://github.com/gorhill/uBlock/commit/b283d6aa33ad27559de961a898f1320d053a7263>

Use the special value `$remove$` to remove an item from the local storage:

    ...##+js(set-local-storage-item, analytics, $remove$)

**Reply by peace2000** [reply]
Yep, seems to work 👍

**Reply by uBlock-user** [reply]
no alias added ?

**Reply by peace2000** [reply]
Alias names for these new scriptlets would be nice indeed.

How about:
`set-cookie` -> `sc`
`trusted-set-cookie` -> `tsc`

`set-local-storage-item` -> `sli`
`trusted-set-local-storage-item` -> `tsli`

**Reply by wavygecko** [reply]
@gorhill I noticed that the examples for this on the Resource Library wiki page use `$remove` instead of `$remove$`. Is that a mistake or is the trailing `$` optional?

**Reply by Yuki2718** [reply]
No and no.

**Reply by gorhill** [reply]
@Yuki2718 Seems to be a mistake, the trailing `$` is missing.

**Reply by Yuki2718** [reply]
Oh, I confused which he thinks is mistake.

**Reply by gorhill** [reply]
Fixed.
